### PR TITLE
add anteHandler for verfying report submiter and feedId

### PR DESCRIPTION
**Pull request by freddyli7**
## Description

- add feed data anteHandler so that only whitelist data provider of the feed is able to submit the report
- add checking for the report submitting to an existing feed based on feed store

### CODE REVIEW

File: x/chainlink/client/cli/txFeed.go [hunk 1]
```diff
@@ -1,14 +1,88 @@
...
+	cmd := &cobra.Command{
+		Use:   "addFeed [feedId] [feedOwnerAddress] [submissionCount] [heartbeatTrigger] [deviationThresholdTrigger] [initDataProviderList]",
+		Short: "Add new feed. Signer must be the existing module owner.",
+		Args:  cobra.MinimumNArgs(6),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			argsFeedId := strings.TrimSpace(args[0])
+			argsFeedOwnerAddr := strings.TrimSpace(args[1])
+			argsSubmissionCount := strings.TrimSpace(args[2])
+			argsHeartbeatTrigger := strings.TrimSpace(args[3])
+			argsDeviationThresholdTrigger := strings.TrimSpace(args[4])
```
Comment (araskachoi) [inline x/chainlink/client/cli/txFeed.go:27]: i dont think we need to trime space for these

File: x/chainlink/client/cli/txFeed.go [hunk 2]
```diff
@@ -1,14 +1,88 @@
...
+			heartbeatTrigger, err := strconv.Atoi(argsHeartbeatTrigger)
+			if err != nil {
+				return err
+			}
+			deviationThresholdTrigger, err := strconv.Atoi(argsDeviationThresholdTrigger)
+			if err != nil {
+				return err
+			}
+
+			argsInitDataProviderList := strings.Split(strings.TrimSpace(argsInitDataProviderListStr), " ")
```
Comment (araskachoi) [inline x/chainlink/client/cli/txFeed.go:43]:
```suggestion
			argsInitDataProviderList := strings.Split(argsInitDataProviderListStr, " ")
```

**Review by araskachoi — APPROVED** [review]
lgtm, just some suggestions

File: x/chainlink/ante/ante.go [hunk 3]
```diff
@@ -132,3 +133,36 @@ func (fd FeedDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, ne
...
+	if len(tx.GetMsgs()) == 0 {
+		return ctx, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "invalid Msg: empty Msg: %T", tx)
+	}
+
+	for _, msg := range tx.GetMsgs() {
+		switch t := msg.(type) {
+		case *types.MsgFeedData:
+			feed := fd.chainLinkKeeper.GetFeed(ctx, t.GetFeedId())
+			if feed.Feed.Empty() {
+				return ctx, sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, "feed not exist")
```
Comment (LuaxY) [inline x/chainlink/ante/ante.go:157]: Error code should be something like `sdkerrors.ErrKeyNotFound`

File: x/chainlink/ante/ante.go [hunk 4]
```diff
@@ -132,3 +133,36 @@ func (fd FeedDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, ne
...
+			}
+			if !(types.DataProviders)(feed.GetFeed().GetDataProviders()).Contains(t.GetSubmitter()) {
+				return ctx, sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, "invalid data provider")
```
Comment (LuaxY) [inline x/chainlink/ante/ante.go:160]: Error code should be something like `sdkerrors.ErrUnauthorized`

**Review by LuaxY — COMMENTED** [review]
LGTM, just some error message are inaccurate I think

File: x/chainlink/client/cli/txFeed.go (hunk 1 above)
Comment (freddyli7) [reply x/chainlink/client/cli/txFeed.go:27]: correct

File: x/chainlink/client/cli/txFeed.go (hunk 2 above)
Comment (freddyli7) [reply x/chainlink/client/cli/txFeed.go:43]: 👍

**Pull request merged**